release/train_tests/benchmark/train_benchmark.py

```diff
@@ -348,8 +351,10 @@ def main():
         factory = ImageClassificationParquetFactory(benchmark_config)
     elif benchmark_config.task == "image_classification_jpeg":
         factory = ImageClassificationJpegFactory(benchmark_config)
+    elif benchmark_config.task == "localfs_image_classification_jpeg":
+        factory = LocalFSImageClassificationFactory(benchmark_config)
     else:
-        raise ValueError
+        raise ValueError(f"Unknown task: {benchmark_config.task}")
```

Comment:
Right now we are adding more "tasks" to support more variants of the image classification task.

I think we can instead expose a task-specific config so that these can be configured via a flag and then we only need a single `ImageClassificationFactory` that instantiates the correct variant.

Example:
```python
task_config.data_format = "jpeg"
task_config.data_location = "local"

if dataloader_type == "ray_data":
    return ImageClassificationRayDataLoaderFactory(..., data_format, data_location)
```

I'm not going to block this PR so we can move fast on the benchmarks, but I can put up a refactor PR after this is merged.